docker-images/kafka/scripts/kafka_config_generator.sh

```diff
@@ -6,11 +6,22 @@
 LISTENERS="REPLICATION://0.0.0.0:9091"
 ADVERTISED_LISTENERS="REPLICATION://$(hostname -f):9091"
 LISTENER_SECURITY_PROTOCOL_MAP="REPLICATION:SSL"
+SASL_ENABLED_MECHANISMS=""
 
 if [ "$KAFKA_CLIENT_ENABLED" = "TRUE" ]; then
   LISTENERS="${LISTENERS},CLIENT://0.0.0.0:9092"
   ADVERTISED_LISTENERS="${ADVERTISED_LISTENERS},CLIENT://$(hostname -f):9092"
   LISTENER_SECURITY_PROTOCOL_MAP="${LISTENER_SECURITY_PROTOCOL_MAP},CLIENT:PLAINTEXT"
```

Comment:
Shouldn't this be set to CLIENT:SASL_PLAINTEXT` when SASL is enabled. Does it work without it? The same should be probably done for the TLS interface (`SASL_SSL` instead of `SSL`).

Reply:
With the current CRD design we can only have a single listener with plain transport and a single one with TLS transport, so there's currently little benefit to parameterising the listener names. AFAICS it would just make constructing the config more complicated. I'll do it if you really want.

Reply:
And does it actually work as it is right now? Because from using this on RHEL I think we actually need this.

Reply:
I've not yet tested it. Can you explain in more detail why it doesn't work?

Reply:
My understanding (and I can be wrong of course) is that you can:
* Use the protocol identifiers directly as the listener names
* Use other names and map them in this protocol map

We cannot really use the first option. Because we have two listeners using `SSL` as protocol. So we will be unable to identify the one which should be used for replication. Even if we used it like this, we still have to change between `PLAINTEXT` and `SASL_PLAINTEXT` and `SSL~ and `SASL_SSL`. Without these SASL will not be enabled.

The second one gives us more flexibility and and allows us to be clear what is the interface purpose etc. And since even with the first one we have to change the types it should not make much difference.

Reply:
@tombentley I think this and the same on line 29 still needs to be fixed before this is merged.

Reply:
@scholzj I saw the usage of `SASL_PLAINTEXT` and `SASL_SSL` in the #795

Reply:
Yes, I think this got changed in #795.

Reply:
Ok, in that case this should be fine.